hashes.lst

```diff
@@ -28,6 +28,8 @@
 # DES) must be last.
 #
 #name    gs_suffix  h_prefix  nrbytes  flags
+yescrypt :          $y$       16       STRONG
+yescrypt :          $7$       16       STRONG
```

Comment:
No, it's not in any OS I'm aware of. However, it's in libsodium (through libsodium's use of a fork of my enhanced scrypt tree that later led to yescrypt), which is a popular library and is available via various language bindings. So such hashes may be in use, but probably not yet by an OS. Since scrypt didn't originally come with an official crypt(3)-style encoding, there are many other encodings also in use. John the Ripper -jumbo includes support for `$7$`, as well as some others.

`$7$` was the encoding I had suggested for scrypt in 2012 or so, and it was in an informal specification document maintained by the author of the eventual scrypt RFC, but ultimately the RFC does not include any crypt(3)-style encoding.

Classic scrypt hashes may also be encoded (and with fewer characters) under the `$y$` prefix (where the variable length integer encoded right after that prefix is the flavor, with single `.` character meaning 0, which is classic scrypt). This is the encoding that `yescrypt_encode_params()` will use if invoked with a params structure with flags=0, requesting classic scrypt.

So for the current yescrypt tree `$7$` is a decoding-only compatibility prefix.

Reply:
Maybe we should not implement `$7$` prefix at all in libcxrypt?

Reply:
Not implementing the `$7$` prefix in libxcrypt at all (at least for now, until/unless we see any demand for it) is OK with me.

Reply:
Stating an argument to not add `$7$` to linxcrypt: we are (in ALT) never intended to add it, it is just side effect of what yescrypt supports, to implement something that is not needed just for the sake symmetry or possibility is strange; it is purely virtual; and scrypt is already properly supported under `$y$` prefix.